Assets/Scripts/Common/Health.cs

```diff
@@ -84,4 +83,10 @@ public class Health : MonoBehaviour {
 		onValueChange.Invoke();
 		return oldValue - newValue;
 	}
-}
\ No newline at end of file
+
+
+	public enum DamageType {
+		Regular,
+		Piercing,
+	}
```

Comment:
I don't think this enum should be nested inside the `Health` class.

Reply:
The damage types are very generic while the death animations are not. What if you get pierced by something that is not a spike?
Maybe we should have something like this instead:
```cs
public enum DamageSource {
	Generic,
	Spikes,
}
```
?